toplevel/g_toplevel.mlg

```diff
@@ -52,7 +52,7 @@ GRAMMAR EXTEND Gram
       | test_show_goal; IDENT "Show"; IDENT "Goal"; gid = natural; IDENT "at"; sid = natural; "." ->
           { Some (VernacShowGoal {gid; sid}) }
       | IDENT "Show"; IDENT "Proof"; IDENT "Diffs"; removed = OPT [ IDENT "removed" -> { () } ]; "." ->
-        { Some (VernacShowProofDiffs (removed <> None)) }
+        { Some (VernacShowProofDiffs (if removed = None then "on" else "removed")) }
```

Comment:
Is it because it can refer to an arbitrary previous version of the document that `VernacShowProofDiffs` has to be in `g_toplevel.mlg` rather than in `vernacexpr.ml` like the other `Show` commands?

Reply:
The code for other commands is called from `Stm`.  `VernacShowProofDiffs` can't call `Stm` to get any previous version at all because that would create a circular linking relationship.  So it's here.